addons/im_livechat/controllers/main.py

```diff
@@ -87,13 +87,17 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, **k
         if request.session.uid:
             user_id = request.env.user.id
             country_id = request.env.user.country_id.id
-        # if geoip, add the country name to the anonymous name
-        elif request.session.geoip:
-            # get the country of the anonymous person, if any
-            country_code = request.session.geoip.get('country_code', "")
-            country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
-            if country:
-                anonymous_name, country_id = _("%s (%s)") % (anonymous_name, country.name), country.id
+        else:
+            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
+            if visitor_sudo:
+                anonymous_name = visitor_sudo.display_name
+            # if geoip, add the country name to the anonymous name
+            if request.session.geoip:
+                # get the country of the anonymous person, if any
+                country_code = request.session.geoip.get('country_code', "")
+                country = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1) if country_code else None
+                if country:
+                    anonymous_name, country_id = "%s (%s)" % (anonymous_name, country.name), country.id
```

Comment:
Don't manually translate country.name
You may also declare the two variables in two lines to ease understanding

Define a "complete_name" or something similar that is either anonymous_name, visitor_sudo.display_name + country thingy. That way it is easier to understand.